bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SortedLedgerStorage.java

```diff
@@ -72,7 +74,13 @@ public void initialize(ServerConfiguration conf,
             checkpointSource,
             checkpointer,
             statsLogger);
-        this.memTable = new EntryMemTable(conf, checkpointSource, statsLogger);
+        if (conf.isEntryLogPerLedgerEnabled()) {
+            this.flushExecutor = OrderedExecutor.newBuilder().numThreads(conf.getNumOfMemtableFlushThreads())
+                    .name("MemtableFlushThreads").build();
+            this.memTable = new EntryMemTableWithParallelFlusher(conf, checkpointSource, statsLogger, flushExecutor);
```

Comment:
it might be better to move flushExecutor to `EntryMemtableWithParallelFlusher` if possible, you can let `MemTable` implements `AutoClose`. #close on `EntryMemtable` is a no-op, while #close on `EntryMemtableWithParallelFlusher` is shutting down the executor. so we don't need this if-else checking in sorted ledger storage.

Reply:
moved